### PR TITLE
enhancement/ERC-responsive-header

### PR DESCRIPTION
**Pull request by sofiaperezm**
# FooCamp - Escuela de Robotica

## Trello ticket
[Enhancement: Responsive en Header](https://trello.com/c/Eb8iRb3i/37-enhancement-responsive-en-header)

## Description

1. Added the collapsing navigation menu component in the header in the `templates/header.hbs` file
2. Added functionality to handle the collapsing navigation menu in the `src/js/handleBurgerMenu.js` file
3. Added the corresponding media queries in `src/scss/layout/_header.scss` file in order to apply responsive design. 

## To reproduce

1. Run the project
2. Open the `http://localhost:8080` page 
3. Check the header component, especially in the small viewports.
4. Interact with the burger menu icon

## Screenshot

Burger menu icon in small viewport
Captura de Pantalla 2021-10-06 a la(s) 11 37 02 p  m
s

Collapsing navigation menu open in small viewport
Captura de Pantalla 2021-10-06 a la(s) 11 37 13 p  m

The collapsing navigation menu feature in action

https://user-images.githubusercontent.com/32691511/136322600-890adc19-b98a-481c-8226-888dda8cacf3.mp4

### CODE REVIEW

File: templates/header.hbs [hunk 1]
```diff
@@ -11,5 +11,16 @@
                 <li class="header__item"><a class="header__link" href="#">DOCUMENTOS</a></li>
                 <li class="header__item"><a class="header__link header__btn" href="#">DONACIONES</a></li>
             </ul>
+            <div>
+                <span class="icon burger__icon">☰</span>
+            </div>
         </nav>
+        <div class="header__menu">
```
Comment (diegoamt) [inline templates/header.hbs:18]: tip: I suggest a better class name like `header__mobile-menu` or similar.

File: templates/header.hbs [hunk 2]
```diff
@@ -11,5 +11,16 @@
...
+            <ul class="links__menu">
```
Comment (diegoamt) [inline templates/header.hbs:19]: following BEM, you should use a class like `header__<block>`. Please replace this class by `header__mobile-links` or similar.

File: templates/link.hbs [hunk 3]
```diff
@@ -0,0 +1 @@
+<li class="header__item"><a class="header__link" href={{linkDestination}}>{{linkText}}</a></li>
```
Comment (diegoamt) [inline templates/link.hbs]: use `href="{{ url }}"` instead of `href={{linkDestination}}`

File: templates/header.hbs [hunk 4]
```diff
@@ -11,5 +11,16 @@
...
+                {{#each navLinks}}
+                    {{{module this}}}
+                {{/each}}
+            </ul>
+            <span class="icon close__icon">✕</span>
```
Comment (diegoamt) [inline templates/header.hbs:24]: please use a different class following BEM, like header__close-icon

**Review by diegoamt — CHANGES_REQUESTED** [review]

**Review by diegoamt — APPROVED** [review]

**Comment by sofiaperezm** [comment]
Please check duplicate  on [PR 27](https://github.com/FooCamp/foocamp-escuela-robotica/pull/27)

**Pull request closed without merging**